addons/sale/models/sale_order.py

```diff
@@ -471,12 +480,24 @@ def _compute_partner_shipping_id(self):
         for order in self:
             order.partner_shipping_id = order.partner_id.address_get(['delivery'])['delivery'] if order.partner_id else False
 
+    @api.depends('company_id')
+    def _compute_has_pricelist(self):
+        for order in self:
+            domain = ['|', ('company_id', '=', False), ('company_id', '=', self.company_id.id)]
+            order.has_pricelist = self.env['product.pricelist'].search(domain)
+
     @api.depends('partner_id')
     def _compute_pricelist_id(self):
         for order in self.filtered(lambda o: o.partner_id):
             order = order.with_company(order.company_id)
             order.pricelist_id = order.partner_id.property_product_pricelist
 
+    @api.depends('pricelist_id', 'company_id')
+    def _compute_currency_id(self):
+        for order in self:
+            order = order.with_company(order.company_id)
```

Comment:
```suggestion
```
Not necessary, you are not reading anything depending on the environment company.